src/compiler/scala/tools/nsc/typechecker/MethodSynthesis.scala

```diff
@@ -76,10 +76,18 @@ trait MethodSynthesis {
     def forwardMethod(original: Symbol, newMethod: Symbol)(transformArgs: List[Tree] => List[Tree]): Tree =
       createMethod(original)(m => gen.mkMethodCall(newMethod, transformArgs(m.paramss.head map Ident)))
 
+    def callStaticsGenericMethodName(name: TermName, returnType: Type)(args: Tree*): Tree = {
+      val method = RuntimeStaticsModule.info.member(name)
+      Apply(TypeApply(gen.mkAttributedRef(method), List(TypeTree(returnType))), args.toList)
+    }
+
+    def callStaticsGenericMethod(name: String, returnType: Type)(args: Tree*): Tree =
+      callStaticsGenericMethodName(newTermName(name), returnType)(args: _*)
+
     def createSwitchMethod(name: Name, range: Seq[Int], returnType: Type)(f: Int => Tree) = {
       createMethod(name, List(IntTpe), returnType) { m =>
         val arg0    = Ident(m.firstParam)
-        val default = DEFAULT ==> Throw(IndexOutOfBoundsExceptionClass.tpe_*, fn(arg0, nme.toString_))
+        val default = DEFAULT ==> callStaticsGenericMethod("aioobe", returnType)(arg0)
```

Comment:
you should add a `val aioobe = "aioobe"` to `nme` in `StdNames` rather than calling `newTermName` every time.